flink-yarn-tests/src/test/java/org/apache/flink/yarn/CliFrontendYarnAddressConfigurationTest.java

```diff
@@ -191,27 +213,47 @@ public void testResumeFromYarnID() throws Exception {
 	@Test
 	public void testResumeFromYarnIDZookeeperNamespace() throws Exception {
 		File directoryPath = writeYarnPropertiesFile(validPropertiesFile);
+
 		// start CLI Frontend
-		TestCLI frontend = new CustomYarnTestCLI(directoryPath.getAbsolutePath());
+		TestCLI frontend = new CustomYarnTestCLI(new Configuration(), directoryPath.getAbsolutePath());
 
-		RunOptions options =
-				CliFrontendParser.parseRunCommand(new String[] {"-yid", TEST_YARN_APPLICATION_ID.toString()});
+		FlinkYarnSessionCli cli = new FlinkYarnSessionCli("y", "yarn");
 
-		frontend.retrieveClient(options);
+		Options options = CliFrontendParser.getRunCommandOptions();
+		cli.addGeneralOptions(options);
+		cli.addRunOptions(options);
+
+		final CommandLine commandLine = CliFrontendParser.parse(options, new String[] {"-yid", TEST_YARN_APPLICATION_ID.toString()}, true);
+
+		RunOptions runOptions = new RunOptions(commandLine);
+
+		frontend.retrieveClient(runOptions);
 		String zkNs = frontend.getConfiguration().getValue(HighAvailabilityOptions.HA_CLUSTER_ID);
 		Assert.assertTrue(zkNs.matches("application_\\d+_0042"));
 	}
 
 	@Test
 	public void testResumeFromYarnIDZookeeperNamespaceOverride() throws Exception {
 		File directoryPath = writeYarnPropertiesFile(validPropertiesFile);
+
+		final Configuration configuration = new Configuration();
+		configuration.setString(YarnConfigOptions.PROPERTIES_FILE_LOCATION, directoryPath.getAbsolutePath());
+
 		// start CLI Frontend
-		TestCLI frontend = new CustomYarnTestCLI(directoryPath.getAbsolutePath());
+		TestCLI frontend = new CustomYarnTestCLI(configuration, directoryPath.getAbsolutePath());
 		String overrideZkNamespace = "my_cluster";
-		RunOptions options =
-				CliFrontendParser.parseRunCommand(new String[] {"-yid", TEST_YARN_APPLICATION_ID.toString(), "-yz", overrideZkNamespace});
 
-		frontend.retrieveClient(options);
+		FlinkYarnSessionCli cli = new FlinkYarnSessionCli("y", "yarn");
```

Comment:
There is redundancy in how `FlinkYarnSessionCli` and `Options` are created. Maybe it makes sense to extract something to instance variables.